var/spack/repos/builtin/packages/py-astropy/package.py

```diff
@@ -60,8 +66,10 @@ class PyAstropy(PythonPackage):
 
     # Optional dependencies
     with when("+all"):
+        depends_on("py-scipy@1.8:", when="@6:", type=("build", "run"))
         depends_on("py-scipy@1.3:", when="@5:", type=("build", "run"))
         depends_on("py-scipy@0.18:", type=("build", "run"))
+        depends_on("py-matplotlib@3.3:", when="@6:", type=("build", "run"))
```

Comment:
`pyproject.toml` indicates there are some versions that should be skipped here:

```
recommended = [
    "scipy>=1.8",
    "matplotlib>=3.3,!=3.4.0,!=3.5.2",
]
```

So you could use comma-separated version ranges (see last example at https://spack.readthedocs.io/en/latest/packaging_guide.html#version-ranges):

```suggestion
        depends_on("py-matplotlib@3.3:3.3,3.4.1:3.5.1,3.5.3:", when="@6:", type=("build", "run"))
```

Reply:
There's a conflict after all the dependencies that handles this since there are other versions of py-astropy that have the same issue